projects/kit/components/slider/slider-readonly.directive.ts

```diff
@@ -16,18 +24,51 @@ const SLIDER_INTERACTION_KEYS = new Set([
  * Native <input type='range' readonly> doesn't work.
  * This directive imitates this native behaviour.
  */
+// @dynamic
 @Directive({
     selector: 'input[tuiSlider][readonly]',
+    providers: [TuiDestroyService],
 })
 export class TuiSliderReadonlyDirective {
     @Input()
     @tuiDefaultProp()
     readonly: '' | boolean = true;
 
+    constructor(
+        @Inject(ElementRef) elementRef: ElementRef<HTMLInputElement>,
+        @Inject(DOCUMENT) documentRef: Document,
+        @Inject(TuiDestroyService)
+        destroy$: Observable<unknown>,
+    ) {
+        const touchStart$ = typedFromEvent(elementRef.nativeElement, 'touchstart', {
+            passive: false,
+        });
+        const touchMove$ = typedFromEvent(documentRef, 'touchmove', {
+            passive: false,
+        });
+        const touchEnd$ = typedFromEvent(documentRef, 'touchend', {
+            passive: true,
+        });
+
+        const shouldPreventMove$ = merge(
+            touchStart$.pipe(
+                tap(e => this.preventEvent(e)),
+                mapTo(true),
+            ),
+            touchEnd$.pipe(mapTo(false)),
+        );
+
+        combineLatest([touchMove$, shouldPreventMove$])
+            .pipe(
+                filter(([_, shouldPreventMove]) => shouldPreventMove),
+                takeUntil(destroy$),
+            )
+            .subscribe(([moveEvent]) => this.preventEvent(moveEvent));
+    }
+
     @HostListener('mousedown', ['$event'])
-    @HostListener('touchstart', ['$event'])
     preventEvent(event: Event) {
-        if (this.readonly === '' || this.readonly) {
+        if (tuiCoerceBooleanProperty(this.readonly) && event.cancelable) {
```

Comment:
`if (event.cancelable && tuiCoerceBooleanProperty(this.readonly)) {`

Reply:
we can't make object destructuring here because function ```ts preventDefault()``` will lose its context and we will get error.

Reply:
but i can change order inside if-condition 👌